functions/supplier/ftp-supplier/src/main/java/org/springframework/cloud/fn/supplier/ftp/FtpSupplierConfiguration.java

```diff
@@ -123,19 +132,24 @@ public Flux<Message<?>> ftpMessageFlux() {
 
 	@Bean
 	@ConditionalOnExpression("environment['file.consumer.mode'] != 'ref'")
-	public Publisher<Message<Object>> ftpReadingFlow() {
+	public Publisher<Message<Object>> ftpReadingFlow(FtpInboundFileSynchronizingMessageSource ftpMessageSource) {
```

Comment:
It was not clear to me at first why we do not pass in the `FtpInboundChannelAdapterSpec ftpMessageSource` and then call `ftpMessageSource.get()`. Is it because the framework will have already called `get()` and pass us the resulting message source?

Reply:
Correct. That’s how I designed SI Java DSL: the `Spec` is a `FactoryBean` by itself, so no need to call it’s result manually and in some cases even dangerous. Perhaps that `get()` has to be hidden somehow, but fully different story…

when we autowire on the other side, we have to deal with the result of the `FactoryBean`. Not sure if Spring describes that somewhere, but I just share our experience what we have so far

Reply:
Thx for the explanation.